src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/Contract.cs

```diff
@@ -7,20 +7,40 @@
 using System;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
 
 namespace Roslyn.Utilities
 {
     internal static class Contract
     {
+        public static Exception Unreachable
+            => ExceptionUtilities.Unreachable;
+
+        public static Exception UnexpectedValue(object? o)
+            => ExceptionUtilities.UnexpectedValue(o);
+
         /// <summary>
         /// Throws a non-accessible exception if the provided value is null.  This method executes in
         /// all builds
         /// </summary>
-        public static void ThrowIfNull<T>([NotNull] T value, string? message = null) where T : class?
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Please add a comment explaining the motivation for this.

Reply:
Added a comment.